src/models/queue.ts

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+type QueueNode<T> = {
+  value: T,
+  next?: QueueNode<T>;
+};
+
+/**
+ * This is a simple queue using an underlying linked list for O(1) enqueue and dequeue operations.
+ * 
+ * @template [T=Object]
+ */
+export default class Queue<T=Object> {
```

Comment:
The implementation seems good to me. But why do we need this? Isn't an array enough?

Reply:
The most simple queue implementation with array would use `array.shift` which is O(n) (depending on the engine, firefox managed to make it O(1)).
This also would not be a problem on the current implementation but the original intent was to use it as a queue to process all transactions (the ones comming from the sync and websocket) and in this case it would be problematic since the array would contain all transactions in the history during it's first processing
Shifting all transactions one by one would be O(n^2) this is why it was discarded.

The load refactor may include this feature of unifying the transaction processing, this is why I chose to maintain this queue approach.

Reply:
Got it. Thanks! We might consider to use the [collections.js](https://www.collectionsjs.com/) library in the future. The downside is to add a dependency. The upside is to have access to efficient and well-tested data structures.

Reply:
I considered using this library but did not want to add a new dependency just for a queue.
But it has some other very interesting data structures like sorted maps which could be very useful